src/App.js

```diff
@@ -1,16 +1,83 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
+import SelectColor from './components/SelectColor';
+
+// iterate through chatMessages and create a list of senders
+const senders = [];
+for (const message of chatMessages) {
+  if (!senders.includes(message.sender)) {
+    senders.push(message.sender);
+  }
+}
+
+// assign first person in senders to local and second to remote (assumes only 2 senders and that the first sender is the local one, wouldn't work for group text!)
+const local = senders[0];
+const remote = senders[1];
+
+// maps through chatMessages and adds local/remote
+const chatMessagesWithLocation = chatMessages.map((message) => {
+  if (message.sender === local) {
+    return { ...message, location: 'local' };
+  } else {
+    return { ...message, location: 'remote' };
+  }
+});
+
 
 const App = () => {
+  const [messageData, setMessageData] = useState(chatMessagesWithLocation);
+
+  const toggleLiked = (id) => {
+    setMessageData((messageData) =>
+      messageData.map((message) => {
+        if (message.id === id) {
+          return { ...message, liked: !message.liked };
+        } else {
+          return message;
+        }
+      })
+    );
+  };
+
+  const getTotalLiked = () => {
+    return messageData.reduce((total, current) => {
+      return total + current.liked;
```

Comment:
Way to use reduce here!

It's cool that you leveraged the knowledge that in JavaScript if you add a boolean to something it gets cast as a number (0 for false, 1 for true). Prefer you use conditional logic to add a number to a number instead.